File: backend/app/models/managed_dashboard.py
"""Pydantic request/response models for RecViz-managed dashboards."""

from __future__ import annotations

from datetime import datetime

from pydantic import Field

from app.models.base import CamelModel


class DashboardCreate(CamelModel):
    name: str = Field(min_length=1, max_length=256)
    description: str = Field(default="", max_length=1024)
    config: dict


class DashboardUpdate(CamelModel):
    name: str | None = Field(default=None, min_length=1, max_length=256)
    description: str | None = Field(default=None, max_length=1024)
    config: dict | None = None


class DashboardResponse(CamelModel):
    id: str
    name: str
    description: str
    config: dict
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/managed-dashboard.ts
import type { DashboardConfig } from '@/types/dashboard-config'

export interface ManagedDashboard {
  id: string
  name: string
  description: string
  config: DashboardConfig
  createdAt: string
  updatedAt: string
}

export interface DashboardCreate {
  name: string
  description: string
  config: DashboardConfig
}

export interface DashboardUpdate {
  name?: string
  description?: string
  config?: DashboardConfig
}
